contracts/Bonding.sol

```diff
@@ -0,0 +1,237 @@
+        uint _bondActivePeriod, 
+        uint _bondToClaimPeriod, 
+        uint _discountNominator,
+        IBondStorage _bondStorage, 
+        AggregatorV3Interface _tokenAggregator,
+        AggregatorV3Interface _gtonAggregator,
+        ERC20 _token,
+        ERC20 _gton,
+        Staking _sgton,
+        bytes memory _bondType
+        ) {
+        bondLimit = _bondLimit;
+        bondActivePeriod = _bondActivePeriod;
+        bondToClaimPeriod = _bondToClaimPeriod;
+        discountNominator = _discountNominator;
+        bondStorage = _bondStorage;
+        tokenAggregator = _tokenAggregator;
+        gtonAggregator = _gtonAggregator;
+        token = _token;
+        gton = _gton;
+        sgton = _sgton;
+        bondType = _bondType;
+    }
+
+    /* ========== MODIFIERS  ========== */
+
+    /**
+     * Mofidier checks if bonding period is open and provides access to the function
+     * It is used for mint function.
+     */
+    modifier mintEnabled() {
+        require(isBondingActive(), 
+            "Bonding: Mint is not available in this period");
+        _;
+    }
+
+    /* ========== CONSTANTS ========== */
+
+    uint constant public discountDenominator = 10000;
+
+    /* ========== STATE VARIABLES ========== */
+
+    bytes public bondType;
+    uint public lastBondActivation;
+    // amount in ms. Shows amount of time when this contract can issue the bonds
+    uint public bondActivePeriod;
+    // Amount in ms. Bond will be available to claim after this period of time
+    uint public bondToClaimPeriod;
+    uint public bondLimit;
+    uint public bondCounter;
+    uint public discountNominator;
+    mapping (uint => BondData) public activeBonds;
+
+    struct BondData {
+        bool isActive;
+        uint issueTimestamp;
+        uint releaseTimestamp;
+        bytes bondType;
+        uint releaseAmount;
+    }
+
+    ERC20 immutable public token;
+    ERC20 immutable  public gton;
+    Staking immutable public sgton;
+    IBondStorage immutable public bondStorage;
+    AggregatorV3Interface public tokenAggregator;
+    AggregatorV3Interface public gtonAggregator;
+
+    /* ========== VIEWS ========== */
+
+    function isActiveBond(uint id) public view returns(bool) {
+        return activeBonds[id].isActive;
+    }
+
+    /**
+     * Function calculates amount of token to be earned with the `amount` by the bond duration time
+     */
+    function getStakingReward(uint amount) public view returns(uint) {
+        uint stakingN = sgton.aprBasisPoints();
+        uint stakingD = sgton.aprDenominator();
+        uint calcDecimals = sgton.calcDecimals();
+        uint secondsInYear = sgton.secondsInYear();
+        uint yearEarn = amount * calcDecimals * stakingN / stakingD;
+        return yearEarn * bondToClaimPeriod / secondsInYear / calcDecimals; 
+    }
+
+    /**
+     * View function returns timestamp when bond period vill be over
+     */
+    function bondExpiration() public view returns(uint) {
```

Comment:
Bond expiration normally refers to the period when a specific bond expires, so just a position of one of the clients. This function better be called bondingWindowEndTimestamp or similar